src/shared/sol-conffile.c

```diff
@@ -58,147 +61,187 @@ sol_conffile_entry_sort_cb(const void *data1, const void *data2)
     return strcasecmp(e1->id, e2->id);
 }
 
+static void
+sol_conffile_set_entry_options(struct sol_conffile_entry *entry, struct sol_json_token options_object)
+{
+    int len;
+    struct sol_json_scanner scanner;
+    struct sol_json_token token, key, value;
+    struct sol_ptr_vector vec_options;
+    enum sol_json_loop_reason reason;
+
+    sol_ptr_vector_init(&vec_options);
+
+    sol_json_scanner_init_from_token(&scanner, &options_object);
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        int key_len, value_len;
+        char *tmp;
+        key_len = sol_json_token_get_size(&key) - 2; // remove quotes from the key.
+        value_len = sol_json_token_get_size(&value); // do not remove quotes from the value. (not everything is a string here)
+        len = key_len + value_len + 2; // plus = and \0
+        tmp = malloc( len );
+        snprintf(tmp, len, "%.*s=%.*s\0", key_len, key.start + 1, value_len, value.start);
```

Comment:
I was looking for a function that did the 'malloc + snprintf' trick, didn't found it, THANKS <3